### PR TITLE
Fix #3935 restore new map background layers on logout

### LINKED ISSUE #3935: logout from new map removes some backgrounds that are not restored after login
### Description
Logout from new map presents some errors.

### In case of Bug (otherwise remove this paragraph)

*Browser Affected*
(use this site: https://www.whatsmybrowser.org/ for non expert users)
- [ ] Internet Explorer
- [x] Chrome
- [ ] Firefox
- [ ] Safari

*Browser Version Affected*

- Indicate the browser version in which the issue has been found

*Steps to reproduce*

- open home of mapstore
- login as admin
- create a new map(you will see multiple background layer to select)
- logout
- login again ( no multiple background layers to select)

*Expected Result*

If you open the background selector the background list is unchanged.

*Current Result*

The backgrounds selector lose all the backgrounds except for the selected one.

### PR DESCRIPTION
**Pull request by kasongoyo**
## Description
restore map backgrounds layers options on logout

## Issues
 - #3935 
 - ...

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)

 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

**What is the current behavior?** (You can also link to an open issue here)
see #3935 

**What is the new behavior?**
restore map backgrounds layers options on logout

**Does this PR introduce a breaking change?** (check one with "x", remove the other)

 - [ ] Yes, and I documented them in migration notes
 - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: web/client/epics/login.js [hunk 1]
```diff
@@ -47,7 +47,7 @@ const refreshTokenEpic = (action$, store) =>
 const reloadMapConfig = (action$, store) =>
     action$.ofType(LOGIN_SUCCESS, LOGOUT)
     .filter(() => pathnameSelector(store.getState()).indexOf("viewer") !== -1)
-    .filter((data) => data.type !== "LOGOUT" ? hasMapAccessLoadingError(store.getState()) : true)
+    .filter((data) => data.type !== "LOGOUT" ? hasMapAccessLoadingError(store.getState()) : pathnameSelector(store.getState()).indexOf("new") === -1)
```
Comment (offtherailz) [inline web/client/epics/login.js]: Why both this and change and the other epic are needed?

File: web/client/epics/login.js [hunk 2]
```diff
@@ -58,6 +58,20 @@ const reloadMapConfig = (action$, store) =>
         return Rx.Observable.of(configureError(e));
     });
 
+const reloadNewMapConfig = (action$, store) =>
```
Comment (offtherailz) [inline web/client/epics/login.js]: I suggest to do everything in hte reloadMapConfig epic

Comment (kasongoyo) [reply web/client/epics/login.js]: @offtherailz I am not sure if you have noticed that the `reloadMapConfig` filter out the `LOGOUT` action while the required behaviour need to work on `LOGOUT` action and specifically the logout that happens on new map context(not all logout). I found out that modifying `reloadMapConfig` to include `LOGOUT` action and the required behaviour was difficult and complicated and I couldn't achieve it as the result I introduced a new epic to tackle only the new behaviour. That was clean and readable than the other way around but I am open if there is a way I could incorporate the behaviour on existing epic, if you can get sometime then we can talk about this challenge

**Comment by tdipisa** [comment]
@kaso please, can you backport this to 2019.02.xx stable branch ?

**Pull request merged**